protocol/encryption/persistence.go

```diff
@@ -727,3 +728,153 @@ func (s *sqliteSessionStorage) Load(id []byte) (*dr.State, error) {
 		return nil, err
 	}
 }
+
+type HRCache struct {
+	GroupId string
+	KeyId   uint32
+	Key     []byte
+	Hash    []byte
+	SeqNo   uint32
+}
+
+// GetCommunityKeyByID retrieves a community key by ID and seqNo. If seqNo = 0, then
+func (s *sqlitePersistence) GetCommunityKeyByID(communityId []byte, keyId uint32, seqNo uint32) (*HRCache, error) {
```

Comment:
We probably want to cover this with indexes, though can be done separately, but maybe add a comment